From 8a13b8a0ee22fa05f282553ec1acfdb291267d24 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 12 May 2021 13:07:05 +0200
Subject: [PATCH] Fix build with musl libc

Use FindIntl cmake modules which will leave Intl_LIBRARIES empty if
libintl is provided by the libc.
---
 CMakeLists.txt            | 1 +
 plugins/nx/CMakeLists.txt | 1 +
 src/CMakeLists.txt        | 3 +--
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb635b05..c187d701 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,6 +52,7 @@ include(GtkUpdateIconCache)
 include(CheckHasModule)
 include(FindPCRE2)
 include(GlibCompileResourcesSupport)
+include(FindIntl)
 
 #---------------------------------------------------------------------
 # Package creation with CMake depends on the value of umask - if permissions
diff --git a/plugins/nx/CMakeLists.txt b/plugins/nx/CMakeLists.txt
index 0106aa73..275a3a6d 100644
--- a/plugins/nx/CMakeLists.txt
+++ b/plugins/nx/CMakeLists.txt
@@ -53,6 +53,7 @@ target_link_libraries(remmina-plugin-nx
     ${REMMINA_COMMON_LIBRARIES}
     ${XKBFILE_LIBRARIES}
     ${LIBSSH_LIBRARIES}
+    ${Intl_LIBRARIES}
     ${X11_X11_LIB})
 
 install(TARGETS remmina-plugin-nx DESTINATION ${REMMINA_PLUGINDIR})
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 08e54043..c670c55d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -207,8 +207,7 @@ target_link_libraries(remmina ${X11_LIBRARIES})
 
 target_link_libraries(remmina ${CMAKE_THREAD_LIBS_INIT})
 
-find_package(Intl)
-if(INTL_FOUND AND "${CMAKE_SYSTEM_NAME}" MATCHES "FreeBSD|OpenBSD|NetBSD")
+if(Intl_FOUND)
 	message(STATUS "${CMAKE_SYSTEM_NAME} detected, building with Intl")
 	include_directories(${Intl_INCLUDE_DIRS})
 	target_link_libraries(remmina ${Intl_LIBRARIES})
-- 
2.31.1

